src/black/strings.py

```diff
@@ -137,6 +137,10 @@ def normalize_string_prefix(s: str, remove_u_prefix: bool = False) -> str:
     assert match is not None, f"failed to match string {s!r}"
     orig_prefix = match.group(1)
     new_prefix = orig_prefix.replace("F", "f").replace("B", "b").replace("U", "u")
+
+    # Python syntax guarantees max 2 prefixes and that one of them is "r"
+    if len(new_prefix) == 2 and "r" == new_prefix[0].lower():
+        new_prefix = new_prefix[::-1]
```

Comment:
Sadly `sorted` isn't quite enough here because of `ur` :/

Reply:
Could use a key like this with `sorted`, I think it's nicer than the reverse but I'm not sure if it reads better overall.
```py
new_prefix = "".join(sorted(new_prefix, key="fubrR".index))
```
`STRING_PREFIX_CHARS` could also be reordered to be used as the index string

Reply:
That could actually be quite fun 😄 but with #2251 we are still second guessing the decisions, so this is a bit blocked.

Reply:
https://github.com/psf/black/pull/2740 has been merged now